### PR TITLE
[release/v2.7] Fix fleet cluster creation to include private registry in spec when legacy/management cluster is detected

### PR DESCRIPTION
**Pull request by Oats87**
## Issue: 
https://github.com/rancher/rancher/issues/39468
 
## Problem
When a v3 (rke1) cluster is used, Rancher deploys the fleet agent (by way of fleet) incorrectly.
 
## Solution
The solution for this is a 5 line change, but to illustrate the actual change being made, I will describe how Fleet-agent is deployed into a Rancher cluster.

Rancher has a controller called the "fleetcluster" controller, which creates corresponding Cluster objects for Fleet. Fleet deploys the fleet-agent into the downstream cluster by way of [manager initiated registration](https://fleet.rancher.io/cluster-overview#manager-initiated-registration). As such, Fleet cluster objects need to be created. The original logic that was implemented for private registry assumed that the private registry information always existed on the `clusters.provisioning.cattle.io/v1` object, which is not the case when dealing with an RKE1 cluster. As such, the logic is now updated to switch the source of the private cluster URL based on the type of cluster detected. 

### Manual Testing
I deployed an RKE1 cluster with a cluster-level private registry, and observed that the fleet agent was pulled from my private registry. 

### Automated Testing
N/A

## QA Testing Considerations
N/A
 
### Regressions Considerations
We should make sure to test fleet-agent success for RKE1 clusters when a private registry is not defined.

### CODE REVIEW

File: pkg/controllers/provisioningv2/fleetcluster/fleetcluster.go [hunk 1]
```diff
@@ -29,6 +30,11 @@ type handler struct {
 	apply         apply.Apply
 }
 
+// Register registers the fleetcluster controller, which is responsible for creating fleet cluster objects.
+// When fleet cluster objects are created, Fleet uses the object to deploy the fleet-agent into the cluster. Notably,
```
Comment (Oats87) [inline pkg/controllers/provisioningv2/fleetcluster/fleetcluster.go:34]: I'm hoping this comment helps future engineers find the source of `fleet-agent` when searching the codebase for it.

**Review by HarrisonWAffel — APPROVED** [review]

File: pkg/controllers/provisioningv2/fleetcluster/fleetcluster.go (hunk 1 above)
Comment (HarrisonWAffel) [reply pkg/controllers/provisioningv2/fleetcluster/fleetcluster.go:34]: Thank you for adding that, it clears up a lot of the confusion I had initially

File: pkg/controllers/provisioningv2/cluster/controller.go [hunk 2]
```diff
@@ -189,7 +200,9 @@ func (h *handler) generateProvisioning(cluster *v3.Cluster, status v3.ClusterSta
 	}, status, nil
 }
 
-func (h *handler) generateCluster(cluster *v1.Cluster, status v1.ClusterStatus) ([]runtime.Object, v1.ClusterStatus, error) {
+// generateLegacyClusterFromProvisioningCluster will generate and return a clusters.management.cattle.io/v3 object based
+// on the passed in.
```
Comment (snasovich) [inline pkg/controllers/provisioningv2/cluster/controller.go:204]: Nit, but looks like it's missing reference to `clusters.provisioning.cattle.io/v1`
```suggestion
// on the clusters.provisioning.cattle.io/v1 object passed in.
```

File: pkg/controllers/provisioningv2/cluster/controller.go [hunk 3]
```diff
@@ -261,6 +274,8 @@ func mgmtClusterName() (string, error) {
 	return name.SafeConcatName("c", "m", rand[:8]), nil
 }
 
+// createNewCluster creates a clusters.management.cattle.io/v3 object based cluster of the
+// clusters.provisioning.cattle.io/v1 object and the spec of the clusters.management.cattle.io/v3 object passed in.
```
Comment (snasovich) [inline pkg/controllers/provisioningv2/cluster/controller.go:278]: Maybe it's just me, but reading through this comment I don't understand what the func is doing. Probably a typo / missing word?

File: pkg/cluster/private_registry.go [hunk 4]
```diff
@@ -21,10 +24,21 @@ func GetPrivateRepoURL(cluster *v3.Cluster) string {
 	return registry.URL
 }
 
+func GetPrivateClusterLevelRepoURL(cluster *v3.Cluster) string {
+	registry := GetPrivateClusterLevelRepo(cluster)
+	if registry == nil {
+		return ""
+	}
+	return registry.URL
+}
```
[the remark this answers is not in the document]
Comment (Oats87) [reply pkg/cluster/private_registry.go]: Removed!

File: pkg/controllers/provisioningv2/cluster/controller.go (hunk 2 above)
Comment (Oats87) [reply pkg/controllers/provisioningv2/cluster/controller.go]: comment corrected!

File: pkg/controllers/provisioningv2/cluster/controller.go (hunk 3 above)
Comment (Oats87) [reply pkg/controllers/provisioningv2/cluster/controller.go]: I read it again and realized I wrote junk, so I fixed it.

**Review by HarrisonWAffel — APPROVED** [review]

**Review by snasovich — APPROVED** [review]

**Review by snasovich — APPROVED** [review]

**Review by HarrisonWAffel — APPROVED** [review]

**Pull request merged**